### ISSUE TITLE
A Service Contract?

### ISSUE THREAD

**Issue by IvanChepurnyi** [Cotya/QueueModule#1]
Would be nice to have an interface for queue, as it is a bad idea to extend always from abstract class.

**Comment by Flyingmana** [comment]
Yeah, I built it just the way I need it currently. For an interface I would hope some input from people who actually know about queues. which then maybe makes also sense if used with an external service for the queue
